modules/apps/portal-cache/portal-cache-impl/src/main/java/com/liferay/portal/cache/internal/dao/orm/EntityCacheImpl.java

```diff
@@ -351,6 +351,8 @@ private void _putResult(
 			return;
 		}
 
+		DBPartitionUtil.populateCompanyId(clazz, baseModel);
```

Comment:
This is wrong, we can not just quietly change the incoming baseModel's company id. As a SharedModel, it must already have a valid companyId in it. We must respect what the caller set it for, rather than swapping it quietly to the threadlocal value.

Reply:
@shuyangzhou this method is only used for the nullModel case (companyId equals 0). This covers the use case when we cache a nullModel an object that exist in another partition. I mark the object with the companyId so when we call getResult from the proper company we know that was cached as nullModel in another company and we return null so we go to the database. Do you have any idea to solve this case?

Reply:
If you leave NullModel without companyId, you won't have this issue.